jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -132,7 +142,40 @@ private static void initEncodings() {
 		if (osEncoding == null)
 			osEncoding = definedOSEncoding;
 	}	
-		
+
+	/*[IF Sidecar18-SE-OpenJ9 | JAVA_SPEC_VERSION >= 11]*/
+	/*
+	 * Return the Charset name of the primary or fallback encoding, if different from the default.
+	 */
+	static String getCharsetName(String primary, String fallback, Charset defaultCharset) {
```

Comment:
Is there any planned usage for `fallback` other than `props.getProperty("file.encoding")` and `null`?
The reason I am asking is that `Charset.defaultCharset()` gets its `Charset` via same system property `file.encoding` as well hence I suspect the fallback is same as defaultCharset which is not necessary.

Reply:
I think Jason is correct. In System, the fallback name is null. The calls from Thread should match the default charset.
I think the fallback parameter should simply be removed.

Reply:
If this method returned the Charset rather than it's name, it would simplify the code where it's used.

Reply:
We can do that for Java 11, but not Java 8 (the `PrintStream` constructor we would want to use was added in Java 10).

Reply:
I'll add more comments to better explain why we need to use file.encoding in Thread.